src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -686,7 +686,10 @@ public void setAllowEmptyCatches(boolean allow) {
     @Override
     public void visitToken(DetailAST ast) {
         final int currentType = ast.getType();
-        if (!isNotRelevantSituation(ast, currentType)) {
+        int childType = 0;
+        if(ast.hasChildren()){
+            childType = ast.getLastChild().getType();}
+        if (!isNotRelevantSituation(ast, currentType) || isArrayInitialization(currentType, childType) ){
```

Comment:
@romani @rnveach There is one more problem right here.
Currently, It is checking if `ARRAY_INIT` `(ARRAY_INIT -> { )` is preceded and followed with whitespace or not. It is not checking whether `RCURLY` is preceded with whitespace or not.
What I am thinking is 
```
 private static boolean isArrayInitialization1(int currentType, int parentType) {
        return (parentType == TokenTypes.ARRAY_INIT || parentType == TokenTypes.ANNOTATION_ARRAY_INIT) &&
                currentType == TokenTypes.RCURLY;
```
and then 
```
if (!isNotRelevantSituation(ast, currentType) || isArrayInitialization(currentType, childType)  || isArrayInitialization1(int currentType, int parentType) )
```
Clearly, this is not the best way to write a check, so I wanted a bit of advice on how to make it check whether `RCURLY` is preceded with whitespace or not.